### ISSUE TITLE
unable to span or stretch across multiple monitors

### ISSUE THREAD

**Issue by bonds** [Plagman/gamescope#737]
This maybe more of a feature request, or an ill conceived hope, but I'm trying to use gamescope to play games stretched across 3 (portrait right oriented) monitors, because it would look awesome. So far I haven't been able to pull it off. I've tried: `gamescope -w 6480 -h 3840 -W 6480 -H 3840 -b thegame` and `gamescope -W 6480 -H 3840 -b thegame` and those just lead to games taking up a 16:9 portion in the middle of my main monitor, leaving black bars of unused space, rather than taking up the whole height of the main monitor (which is what I get if I run the game without gamescope), and rather than spanning all 3 monitors and taking up all of their space, like I really want.

Am I doing it wrong? Or is this not supported? I'm running the latest Manjaro with Gnome.

**Comment by KagoraTheGrindcoreAddict** [comment]
Does the game not support the resolution or aspect ratio?
Works fine for me even across different resolution monitors (2560+3440+2560)x1440
`gamescope -b -W 8560 -H 1440 -r 400 -o 30 -- game`
![wide](https://user-images.githubusercontent.com/103851071/211180031-0e597851-6841-401a-a4a5-9087c26f1788.jpg)

**Comment by bekopharm** [comment]
Hej @bonds ,
doing that all the time (also with vscode and other apps). Not sure on this but afaik it's Gnome that resizes the window back to primary screen dimensions on launch and I don't think that `gamescope` can do much about it.

You can however drag the window to (or outside of) the upper left corner by pressing left mouse button + superkey on the game window and resize it from there again using the usual window resize function with your mouse.

You may also script that so some extend with `wmctrl` to get the ratio right. That doesn't take negative numbers tho and oddly enough it's the Gnome dock that may prevent the window from sliding "under it" but you can still move the window manually after resizing it. Just take care to not hit a "smart border" or the window resizes automatically again!

Sometimes `wmctrl` does not work or I have to run the command _twice_. I found no pattern to this yet but I do _have to_ move the window to the upper left corner first and it must no longer be `maximized`. Probably some Wayland quirk with `wmctrl`.

Example with the game X4 to resize it after launch (without gamescope):

> wmctrl -x -r X4.X4 -e 0,0,0,5760,1220

Example with Rebel Galaxy Outlaw with gamescope:

> wmctrl -x -r gamescope.gamescope -e 0,0,0,5760,1220

(I always add ~20 pixels on the height for the window title bar. That can also be forcefully removed but that's more work)

![rebel-galaxy-outlaw-with-the-little-one](https://user-images.githubusercontent.com/5016428/223078377-c793049f-54c2-46b2-a670-b243353ebfc0.jpg)

If some game really refuses to cooperate you may consider setting up a virtual display. I described the process for X11 and an Intel iGPU here: https://beko.famkos.net/2021/05/31/extreme-multihead-for-gaming-on-linux-pc/

This applies for portrait mode too. Did that for some time too 🤓

**Comment by bonds** [comment]
Thanks @bekopharm ! I'm on Manjaro with Gnome which comes with Wayland which means wmctrl doesn't work and AFAIK there's no equivalent tool for Wayland, where you can resize windows from the command line. I didn't realize Gnome does the resizing...that makes sense now why gamescope can't do anything about it.

I'll consider setting up a virtual display. I rather like the snapping to individual displays outside of when I'm gaming, so I'll need to figure out how to get a similar effect if I go for the one big display option.

Appreciate you sharing what you know and ideas for me to try!

**Comment by Riverside96** [comment]
@bonds did you have any success with this? I'm using 3 portrait monitors on wayland also.

**Comment by bonds** [comment]
No. I ended up writing a script that uses wmctl to maximize across my
monitors and I assigned that to a keyboard shortcut.

https://github.com/bonds/dotfiles/blob/main/bin/linux/maximize_across_multiple_monitors

**Comment by Riverside96** [comment]
Nice thanks! I couldn't see any wms configured in your dots. I take it you switched back to x11 for wmctrl this to work?

**Comment by Riverside96** [comment]
Nice thanks! I couldn't see any wms configured in your dots. I take it you switched back to x11 for wmctrl this to work?

**Comment by bonds** [comment]
I’m running the latest Manjaro with Gnome, mutter, and Wayland. I gather wmctrl added support for moving around Wayland windows since I last tried it.

**Comment by bekopharm** [comment]
I found out meanwhile what makes this work.

Some time ago `gamescope` was always run via `XWayland` so `wmctrl` could also see and manipulate it's window.

A more uptodate `gamescope` seems to run directly via `Wayland` now though. In that case `wmctrl` does not see the game window.

To make this work again it has to be set to use `X11` again to it is run via `XWayland` on `Wayland` in the end. Here is an example how I do this e.g. as start option for `steam` games:

> SDL_VIDEODRIVER=x11 gamescope -h 1200 -w 5760 -H 1200 -W 5760 -b -e -- mangohud %command% -force-vulkan

Or, in the case of OP:

> SDL_VIDEODRIVER=x11 gamescope -h 1200 -w 5760 -H 1200 -W 5760 -b -- thegame

This can btw be done more comfortable in a more recent build of `lutris` with a GUI:

![grafik](https://github.com/ValveSoftware/gamescope/assets/5016428/7e60c019-0094-4a3e-9bfe-64591f19f396)

## Virtual monitors ramblings

I recently tried again to create virtual monitors too but this appears to be as dire as before on `Wayland`. Worse, because `xrandr` does provide us with all the bits but _no_ windowmanager I tried - be it on `X11` or `Wayland` does make use of it, what is basically a feature designed for modern "MST" displays. See https://keithp.com/blogs/MST-monitors/ for more.

![grafik](https://github.com/ValveSoftware/gamescope/assets/5016428/1dda1476-f99f-449c-8790-891665ecfde6)

My workaround for `X11` with an Intel GPU (iGPU is enough) still works: https://beko.famkos.net/2021/05/31/extreme-multihead-for-gaming-on-linux-pc/ - It can not be used with an all AMD-system though. I recently switched so I can tell - and while `amdgpu` does in theory also have a virtual display mode it also does disable the physical outputs :( I could not get it to work with my amdgpu iGPU as well - the option was simply ignored and I have no idea why.

It _should_ also be possible to use `Xinerama` with `X11` again as this was done years ago, but this does need a detailed and fine tuned X config, which ssssucks to get right in the first place.

https://github.com/DisplayLink/evdi is in theory capable of doing this too but I didn't bother with this again. It's compile time which I don't exactly fancy for driving a display. I'll wait until this is more mainstream.

So… until that: More `gamescope`. It's the best thing since sliced bread :nerd_face:

**Comment by zastrixarundell** [comment]
A dirty work around on KDE for a 3 monitor setup:

* Start the game with: `gamescope -w 5760 -h 1080 -W 5760 -H 1080 --`  (you CAN use -b but no need)
* ALT+F3 to enter window menu on every step from now-on
* Disable titlebar and frame
* Move to fit to bottom (it should snap to the bottom)
* Enable keep above others

Only issue is that it's not actually vsynced as I can see so you might have a stutter from time to time on a non-VRR monitor.

**Comment by bekopharm** [comment]
tl;dr: Wayland can not do this at the moment and `gamescope` is the way to move forward right now.

> https://www.reddit.com/r/gnome/comments/1ei836j/comment/lg97guc/
